pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3770,6 +3770,21 @@ public boolean isDispatcherPauseOnAckStatePersistentEnabled() {
         return b == null ? false : b.booleanValue();
     }
 
+    @Override
+    public void updateDispatchPauseOnAckStatePersistentEnabled() {
+        super.updateDispatchPauseOnAckStatePersistentEnabled();
+        // Trigger new read if subscriptions has been paused before.
+        if (!brokerService.getPulsar().getConfiguration().isDispatcherPauseOnAckStatePersistentEnabled()) {
+            updateDispatchPauseOnAckStatePersistentEnabled();
+            getSubscriptions().forEach((sName, subscription) -> {
+                if (subscription.getDispatcher() == null) {
+                    return;
+                }
+                subscription.getDispatcher().afterAckMessages(null, 0);
```

Comment:
It's not a good idea to use a dummy message acknowledgment for resuming the read.

I think we should add a new method, e.g. `checkAndResumeIfPaused()`.

Reply:
Sure, added a new method `checkAndResumeIfPaused` to trigger a new read if needed